match/match.py

```diff
@@ -0,0 +1,81 @@
+import json
+
+# Get the userID and University name
+USER_ID = "UID3"  # TODO: connect to db
+USER_UNI = "UCLA"  # TODO: connect to db
+
+
+def main():
+    # open the file
+    with open("data.json", "r") as read_file:  # TODO: connect to db
+        data = json.load(read_file)
+
+    # provided data from the user: interests, name, [previous matches]
+    user_data = data["Universities"][USER_UNI][USER_ID]
+
+    # all users of the university
+    uni_users = data["Universities"][USER_UNI]
+
+    # Keep track of points for each user
+    match_points = {}
+
+    print(user_data["PreviousMatches"])
+
+    max_user = -1
+    user_ids = []
+    for i in uni_users:
+        max_user += 1
+        user_ids.append(i)
+    print(max_user)
+
+    """
+    This function loops over the data of user's university and 
+    assigns points for each of the other users 
+    """
+
+    def match():
+        points = 0  # increment when interests match
+        for user in uni_users:  # loop over the whole json file. i is user id
+            if USER_ID != user:  # exclude main user's data for the next loop
+                # check if previously matched
+                matched_before = previously_matched(user)
+                if not matched_before:
+                    print("not matched")
+                    # loop over the other user's interests
+                    for j in data["Universities"][USER_UNI][user]["Interests"]:
+                        # check if interests match
+                        try:
+                            if user_data["Interests"][j] == 1 and uni_users[user]["Interests"][j] == 1:
```

Comment:
A couple things:
1. the tags won't necessarily correspond with each other in the same order (e.g. the interest "music" might be in location 0 for user A and in location 3 for user B so you'll have to cross reference the tag with all the tags the other user has for each tag

2. the tag values are strings, so `user_data["Interests"][j]` will be equal to something like "music" or "cooking"

I'd recommend a nested loop where you loop through one user's interests and check if they equal any of the other user's tags, and if so, `points += 1`